### PR TITLE
[dynamo][refactor] Refactor Wrap HOP to reuse it for invoke_subgraph

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137808
* #138036
* #137965
* #138031
* #137557
* #137538
* #138424
* __->__ #137555

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Comment by anijain2305** [comment]
@zou3519 This is ready for review.

**Review by zou3519 — APPROVED** [review]

File: torch/_dynamo/variables/higher_order_ops.py [hunk 1]
```diff
@@ -1478,10 +1478,20 @@ def call_function(
 
 
 class WrapHigherOrderVariable(TorchHigherOrderOperatorVariable):
+    def install_subgraph_in_output_graph(
```
Comment (zou3519) [inline torch/_dynamo/variables/higher_order_ops.py:1481]: Wrap is the base for some other HOPs too, right? (like checkpoint?).

We should probably give it a better name at some point. "Wrap" is basically a HOP that ends up executing the subgraph you give it without changing the calling convention of that subgraph. Elias was also building a HOP on top of Wrap (called invoke_quant)

**Pull request closed without merging**